Merge feature branch

davidauza-engineer opened the pull request:
Please let us know your comments about our work.

Commit:
files and folder included

Commit:
.rspec file created

Commit:
rewrite the tests and file created

Commit:
spec helper file tests

Commit:
board file created

Commit:
Fix the diagonal check. Grid is not upside down. Add smiley faces

Commit:
fixes the diagonal check

Commit:
readme file created

Commit:
rubocop and stickler added

eng-eng11 committed:
Update README.md

davidauza-engineer committed:
Update README.md

stickler-ci committed:
Fixing style errors.

davidauza-engineer committed:
Update .stickler.yml

davidauza-engineer committed:
Update .rubocop.yml

davidauza-engineer committed:
Add main.rb

davidauza-engineer committed:
Relocate main.rb

stickler-ci committed:
Fixing style errors.

davidauza-engineer committed:
Fix style issues

davidauza-engineer committed:
Merge branch 'feature' of github.com:falguniislam/RUBY-Testing-Your-Ruby-Code---Tic-Tac-Toe into feature

davidauza-engineer committed:
Update file

davidauza-engineer committed:
Update files

davidauza-engineer committed:
Update main.rb

stickler-ci committed:
Fixing style errors.

davidauza-engineer committed:
Merge branch 'feature' of github.com:falguniislam/RUBY-Testing-Your-Ruby-Code---Tic-Tac-Toe into feature

davidauza-engineer committed:
Update spec_helper.rb

stickler-ci committed:
Fixing style errors.

davidauza-engineer committed:
Add tic_tac_toe_spec.rb

davidauza-engineer committed:
Update .rspec

davidauza-engineer committed:
Update board.rb

davidauza-engineer committed:
Update game.rb

davidauza-engineer committed:
Update player.rb

davidauza-engineer committed:
Remove lib folder

stickler-ci committed:
Fixing style errors.

davidauza-engineer committed:
Update main.rb

davidauza-engineer committed:
Update spec_helper

davidauza-engineer committed:
Update tic_tac_toe_spec.rb

davidauza-engineer committed:
Update .rspec

davidauza-engineer committed:
Update game.rb

davidauza-engineer committed:
Update palyer.rb

stickler-ci committed:
Fixing style errors.

davidauza-engineer committed:
Resolve conflicts

stickler-ci committed:
Fixing style errors.

davidauza-engineer committed:
update tic_tac_toe_spec.rb

davidauza-engineer committed:
test commit

davidauza-engineer committed:
Fix Style errors

davidauza-engineer committed:
Update README.md

davidauza-engineer committed:
Remove .idea folder

davidauza-engineer committed:
Merge branch 'feature' of github.com:falguniislam/RUBY-Testing-Your-Ruby-Code---Tic-Tac-Toe into feature

stratospherique commented on spec/tic_tac_toe_spec.rb:
```diff
@@ -0,0 +1,46 @@
+require './game.rb'
+
+describe Game do
+  subject(:game) { Game.new }
```
- This line is causing interactions with the user when we run **rspec**
- Use let to define test cases variables

davidauza-engineer replied:
Done, thank you so much :)

stratospherique commented on spec/tic_tac_toe_spec.rb:
```diff
@@ -0,0 +1,46 @@
+      expect(game.over?).to be true
+    end
+  end
+end
```
- You should be testing all the winning conditions :nerd_face:

davidauza-engineer replied:
Added one case for every winning condition as requested 👍

stratospherique reviewed (changes requested):
### Good start team :clap:. However, the specified tests doesn't meet with the project's requirements. check below :arrow_down: 

* [x] All the public methods of the classes implemented in project OOP are tested
   - not all public methods are tested in your specs. :no_entry_sign: 
* [x] The specs should cover positive and negative scenarios Example: there is a spec for correct usage of method \[e.g. “it “creates object if all required attributes are passed”\] and also there is spec for wrong usage of method \[e.g. “it “does not create object if required attributes are missing”\].

* [x] Specs for each method should be wrapped in one `describe` block. As in example here https://github.com/rubocop-hq/rspec-style-guide#empty-lines-around-examples .
* [x] You should have one `it` block for each expected behaviour.

* [x] to define a variable for the test cases, students should [Use `let` instead](https://github.com/rubocop-hq/rspec-style-guide#let-blocks).
 
###### PS: check the code comment below :arrow_down: 

### That's it :relaxed: . When finished, kindly resubmit another code review request :pray: 

happy coding :v: 
**Ahmed Mahfoudh**

davidauza-engineer committed:
Set up RSpec

davidauza-engineer committed:
Add draw_board test

davidauza-engineer committed:
Add update_field_values test for board.rb

davidauza-engineer committed:
Fix style error 2

davidauza-engineer committed:
Fix Style errors

davidauza-engineer committed:
Update tests for board.rb

davidauza-engineer committed:
Update tests for board.rb

davidauza-engineer committed:
Add tests for game.rb

davidauza-engineer committed:
Add tests for Player class

stickler-ci committed:
Fixing style errors.

davidauza-engineer commented:
We made all the requested changes, please let us know if everything is OK now. Thanks in advance :)

Ispirett reviewed (changes requested):
## Status :changes required
Great job writing the tests @davidauza-engineer @falguniislam and implementing changes :+1:  :100: 

Unfortunately, im getting a lot broken test and  other bugs :bug: 

- [x] I should be  able to run rspec from the main folder. When I do the tictactoe game runs
- [x] You should prefix describe classes with like `**RSpec.describe player do**`
- [x] errors bellow when running your tests
rspec ./spec/tic_tac_toe_spec.rb:11 # Board#draw_board draw board appropriately
rspec ./spec/tic_tac_toe_spec.rb:31 # Game#play_turn sets a turn adequately returning 0 for success
rspec ./spec/tic_tac_toe_spec.rb:37 # Game#populate_display populates the display appropriately returning 0 for success
rspec ./spec/tic_tac_toe_spec.rb:43 # Game switch_player successfully switch the player by returning 0 for success
rspec ./spec/tic_tac_toe_spec.rb:49 # Game over? successfully finish the game with the first winning condition
rspec ./spec/tic_tac_toe_spec.rb:53 # Game over? successfully finish the game with the second winning condition
rspec ./spec/tic_tac_toe_spec.rb:57 # Game over? successfully finish the game with the third winning condition
rspec ./spec/tic_tac_toe_spec.rb:61 # Game over? successfully finish the game with the fourth winning condition
rspec ./spec/tic_tac_toe_spec.rb:65 # Game over? successfully finish the game with the fifth winning condition
rspec ./spec/tic_tac_toe_spec.rb:69 # Game over? successfully finish the game with the sixth winning condition
rspec ./spec/tic_tac_toe_spec.rb:73 # Game over? successfully finish the game with the seventh winning condition
rspec ./spec/tic_tac_toe_spec.rb:77 # Game over? successfully finish the game with the eighth winning condition
rspec ./spec/tic_tac_toe_spec.rb:81 # Game over? continue the game if the given array is not one from the winning options

davidauza-engineer committed:
Add RSpec prefix

davidauza-engineer committed:
Fix bugs

stickler-ci committed:
Fixing style errors.

davidauza-engineer commented:
We modified the necessary files to successfully run rspec from the main folder, prefixed the describe classes as requested and fixed all the errors:

<img width="1280" alt="Screen Shot 2019-10-25 at 12 16 01 PM" src="https://user-images.githubusercontent.com/47565678/67590654-38c61d80-f721-11e9-904a-a574b2c2deb3.png">

We'll be waiting for your feedback :)

wathigo reviewed (approved):
### Hi @davidauza-engineer  and @falguniislam  :wave:  
Great work implementing the requested changes.

**You may move on :next_track_button:**  

_Have a great weekend :smile: 
Reviewed by: **Simon Wathigo**_

davidauza-engineer commented:
Hi @stratospherique , @Ispirett  and @wathigo. Thank you so much for your comments and those great reviews 👍
